lib/networkx/graph.rb

```diff
@@ -117,5 +117,216 @@ def remove_nodes(nodes)
+      raise KeyError, 'The given edge is not a valid one.' unless @adj[node_1].key?(node_2)
+      @adj[node_1].delete(node_2)
+      @adj[node_2].delete(node_1) if node_1 != node_2
+    end
+
+    # Removes multiple edges from the graph
+    #
+    # @example
+    #   graph.remove_edges([%w[Noida Bangalore], %w[Bangalore Chennai]])
+    #
+    # @param edges [Array<Object>] the array of edges to be removed
+    def remove_edges(edges)
+      case edges
+      when Array, Set
+        edges.each { |node_1, node_2| remove_edge(node_1, node_2) }
+      else
+        raise ArgumentError, 'Expected Arguement to be Array or Set of edges, '\
+                             "received #{edges.class.name} instead."
+      end
+    end
+
+    # Adds weighted edge
+    #
+    # @example
+    #   graph.add_weighted_edge('Noida', 'Bangalore', 1000)
+    #
+    # @param node_1 [Object] the first node of the edge
+    # @param node_2 [Object] the second node of the edge
+    # @param weight [Integer] the weight value
+    def add_weighted_edge(node_1, node_2, weight)
+      add_edge(node_1, node_2, weight: weight)
+    end
+
+    # Adds multiple weighted edges
+    #
+    # @example
+    #   graph.add_weighted_edges([['Noida', 'Bangalore', 1000],
+    #                             ['Noida', 'Nagpur', 1000]])
+    #
+    # @param edges [Array<Object, Object>] the array of edges
+    # @param weight [Array<Integer>] the array of weights
+    def add_weighted_edges(edges, weights)
+      raise ArgumentError, 'edges and weights array must have equal number of elements.'\
+                           unless edges.size == weights.size
+      raise ArgumentError, 'edges and weight must be given in an Array.'\
+                           unless edges.is_a?(Array) && weights.is_a?(Array)
+      (edges.transpose << weights).transpose.each do |node_1, node_2, weight|
+        add_weighted_edge(node_1, node_2, weight)
+      end
+    end
+
+    # Clears the graph
+    #
+    # @example
+    #   graph.clear
+    def clear
+      @adj.clear
+      @nodes.clear
+      @graph.clear
+    end
+
+    # Checks if a node is present in the graph
+    #
+    # @example
+    #   graph.node?(node_1)
+    #
+    # @param node [Object] the node to be checked
+    def node?(node)
+      @node.key?(node)
+    end
+
+    # Checks if the the edge consisting of two nodes is present in the graph
+    #
+    # @example
+    #   graph.edge?(node_1, node_2)
+    #
+    # @param node_1 [Object] the first node of the edge to be checked
+    # @param node_2 [Object] the second node of the edge to be checked
+    def edge?(node_1, node_2)
+      return true if @nodes.key?(node_1) && @adj[node_1].key?(node_2)
+      false
+    end
+
+    # Gets the node data
+    #
+    # @example
+    #   graph.get_node_data(node)
+    #
+    # @param node [Object] the node whose data is to be fetched
+    def get_node_data(node)
+      return @nodes[node] if @nodes.key?(node)
```

Comment:
Again, the above `node?` method could be used rather than re-writing `@nodes.key?(node)`